### PR TITLE
Support unicast in ibverbs receiver paths

### PR DESCRIPTION
**Pull request by bmerry**
This includes
- library APIs
- spead2_recv and spead_recv.py test programs
- spead2_bench and spead2_bench.py benchmark programs
- mcdump
There are also a few tweaks to interpretation of arguments in the non-verbs code-paths for consistency.

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -3,10 +3,19 @@ Changelog
...
+  number.
+- Support unicast addresses in a few APIs that previously only accepted
+  multicast addresses; in most cases the unicast address must match the
+  interface address.
 - Add missing ``<map>`` include to ``<spead2/recv_heap.h>``.
 - Show the values of immediate items in :program:`spead2_recv`.
 - Fix occasional crash when using thread pool with more than one thread
   together with ibverbs.
+- Fix bug in mcdump causing it to hang if the arguments couldn't be parsed
+  and capturing to file.
```
Comment (sratcliffe) [inline doc/changelog.rst]: Couldn't parse this - are we saying that arguments handling is fixed; and we can now capture to a file ?

File: doc/py-ibverbs.rst [hunk 2]
```diff
@@ -8,9 +8,9 @@ of libibverbs.
 
 There are a number of limitations in the current implementation:
 
- - Only IPv4 is supported
- - VLAN tagging, IP optional headers, and IP fragmentation are not supported
- - Only multicast is supported.
+ - Only IPv4 is supported.
```
Comment (sratcliffe) [inline doc/py-ibverbs.rst]: There is a bunch of IPv6 support in the recv libraries - is this statement specific to using verbs ?

File: doc/changelog.rst (hunk 1 above)
Comment (bmerry) [reply doc/changelog.rst]: It was only broken when capturing to file. I agree it's not the clearest wording - I'll try to improve it.

File: doc/py-ibverbs.rst (hunk 2 above)
Comment (bmerry) [reply doc/py-ibverbs.rst]: Correct.

File: doc/changelog.rst (hunk 1 above)
Comment (bmerry) [reply doc/changelog.rst]: Does it look better now?

**Pull request merged**